clientapi/routing/register.go

```diff
@@ -100,6 +100,11 @@ func newSessionsDict() *sessionsDict {
 	}
 }
 
+// AddCompletedSessionStage records that a session has completed an auth stage.
+func AddCompletedSessionStage(sessionID string, stage authtypes.LoginType) {
```

Comment:
`func (d *sessionsDict) AddCompletedStage` already does the same thing and also prevents duplicate stages being added; I think we should use that method instead.